### ISSUE TITLE
Data shuffling

### ISSUE THREAD

**Issue by tboo** [uber/ludwig#185]
Hi -
I get that one can define data splits in the CSV (e.g. what is used for training, validation and testing) or if no split column is provided the splits and their respective defined fractions are allocated randomly. If the latter is the case are these allocations shuffled after each epoch? E.g. is the training subset always the same the data after initial assignment? I cannot find a parameter to switch shuffling on or off.
Sorry for this somewhat naive question, this is probably visible somewhere inside the code but spotting a feature like that is beyond my skills :(

**Comment by w4nderlust** [comment]
@tboo the split is done once before training, then data within the training split is shuffled each epoch, the subset is always the same, the order in which it is presented changes.
Would you like to turn shuffling off? In that case I may add a training parameter for that, like `shuffle: True` by default that you can set to false.

**Comment by tboo** [comment]
Thanks for the clarification. I am not intending to turn shuffling off. In fact the 'opposite' is the case. My question was if it is possible to shuffle the splits after each epoch, meaning that the validation and training data set is different at each epoch.

The reason I am asking is that my image classification reaches 90%+ accuracy on test and validation data during training but if I use the obtained model for prediction of previously unseen data my accuracy drops to 70%. I wonder if this is because the validation set is not representative enough and I thought selecting new validation data at each epoch would somehow compensate for that. But the key is likely to increase the sample size/split fraction of the validation data. I will play around with that.

**Comment by w4nderlust** [comment]
@tboo I see. Leaking datapoints from the validation set or the test set into the training set is a bad practice, it will increase your numbers, but those numbers will be meaningless and will not improve your performance on additional held out data, it will probably decrease it because it would overfit.
The reason for your performance may have several explanations, among with a consistent different in the distribution between your original data and the held out one. If that's the case, probably spending some effort tagging more data from the same distribution of the held out would probably be a good idea.
Anyway, this is not an issue, but a discussion, I'm closing it and and encourage you to open a discussion on Gitq if you want to continue it further.
